src/RefractDataStructure.cc

```diff
@@ -68,9 +68,46 @@ namespace drafter {
         }
     }
 
+    static mson::BaseTypeName NamedTypeFromElement(const refract::IElement* element) {
+        refract::TypeQueryVisitor type;
+        type.visit(*element);
+
+        switch (type.get()) {
+            case refract::TypeQueryVisitor::Boolean:
+                return mson::BooleanTypeName;
+
+            case refract::TypeQueryVisitor::Number:
+                return mson::NumberTypeName;
+
+            case refract::TypeQueryVisitor::String:
+                return mson::StringTypeName;
+
+            case refract::TypeQueryVisitor::Array:
+                return mson::ArrayTypeName;
+
+            case refract::TypeQueryVisitor::Object:
+                return mson::ObjectTypeName;
+
+            default:
+                return mson::UndefinedTypeName;
+        }
+
+        return mson::UndefinedTypeName;
+    }
+
     template<typename T>
     static mson::BaseTypeName GetType(const T& type) {
-        return type.typeDefinition.typeSpecification.name.base;
+        mson::BaseTypeName nameType =  type.typeDefinition.typeSpecification.name.base;
```

Comment:
Extra space after `=`. I don't know if our style rules will fix this or not when implemented.